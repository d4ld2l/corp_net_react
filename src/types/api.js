import type { AxiosPromise } from 'axios'

import type {
  BirthDatesNearestRaw,
  CandidateRaw,
  CandidateWithIncludesRaw,
  EmployeesRaw,
  EmployeeRaw,
  EventsRaw,
  EventRaw,
  EventTypesRaw,
  ParticipantsRaw,
  UserRaw,
  MilestonesGroupRaw,
  MilestoneTemplateRaw,
  NewRaw,
  NewsRaw,
  NewsCategoriesRaw,
  ProfileRaw,
  ProfileRoleRaw,
  RecruiterRoleInfoRaw,
  SurveysRaw,
  SurveyRaw,
  TagsRaw,
  VacancyRaw,
  VacanciesRaw,
  CandidateVacancyRaw,
  ResumeRaw,
  AdditionalContactRaw,
  ResumeWorkExperienceRaw,
  ResumeRecommendationRaw,
  LanguageSkillRaw,
  LanguageRaw,
  LanguageLevelRaw,
  ResumeEducationRaw,
  EducationLevelRaw,
  CandidatePermittedAttrsRaw,
  ResumeSourceRaw,
  ServicesRaw,
  ServiceRaw,
  BidsRaw,
  BidRaw,
} from './raws'

import type { DepartmentsRaw } from '../presenters/departments'

export type Api = {
  birthdates: {
    nearest: () => AxiosPromise<BirthDatesNearestRaw>,
  },
  candidates: {
    all: () => AxiosPromise<Array<CandidateWithIncludesRaw>>,
    select: (params: string) => AxiosPromise<Array<CandidateWithIncludesRaw>>,
    search: (params: CandidatePermittedAttrsRaw) => AxiosPromise<Array<CandidateWithIncludesRaw>>,
    advancedSearch: (
      params: CandidatePermittedAttrsRaw
    ) => AxiosPromise<Array<CandidateWithIncludesRaw>>,
    update: (
      id: number,
      params: CandidatePermittedAttrsRaw
    ) => AxiosPromise<CandidateWithIncludesRaw | { success: false }>,
    findById: (id: number) => AxiosPromise<CandidateWithIncludesRaw>,
    whereGroupId: (id: number) => AxiosPromise<Array<CandidateWithIncludesRaw>>,
    create: (
      params: CandidatePermittedAttrsRaw
    ) => AxiosPromise<CandidateWithIncludesRaw | { success: false }>,
    parsing: (params: {
      source: string,
      file: string,
    }) => AxiosPromise<CandidateWithIncludesRaw | { success: false }>,
  },
  news: {
    create: (params: Object) => AxiosPromise<NewRaw>,
    all: () => AxiosPromise<NewsRaw>,
    where: (params: { params: * }) => AxiosPromise<NewsRaw>,
    findById: (id: number) => AxiosPromise<NewRaw>,
    categories: () => AxiosPromise<NewsCategoriesRaw>,
  },
  employees: {
    findById: (id: number) => AxiosPromise<EmployeeRaw>,
    search: (params: { params: * }) => AxiosPromise<EmployeesRaw>,
    all: () => AxiosPromise<EmployeesRaw>,
  },
  events: {
    all: () => AxiosPromise<EventsRaw>,
    findById: (id: number) => AxiosPromise<EventRaw>,
    types: () => AxiosPromise<EventTypesRaw>,
    create: (params: Object) => AxiosPromise<EventRaw>,
    update: (id: number, params: Object) => AxiosPromise<EventRaw>,
    delete: (id: number) => AxiosPromise<{}>,
    searchParticipants: (q: string) => AxiosPromise<ParticipantsRaw>,
  },
  auth: {
    signin: (params: Object) => AxiosPromise<UserRaw>,
    logout: () => AxiosPromise<UserRaw>,
  },
  profiles: {
    me: () => AxiosPromise<ProfileRaw>,
  },
  permissions: {
    my: () => AxiosPromise<ProfileRoleRaw>,
  },
  surveys: {
    all: () => AxiosPromise<SurveysRaw>,
    findById: (id: number) => AxiosPromise<SurveyRaw>,
    search: (q: string) => AxiosPromise<SurveysRaw>,
    save: (params: Object) => AxiosPromise<SurveyRaw>,
  },
  users_admin: {
    where: (params: { params: * }) => AxiosPromise<RecruiterRoleInfoRaw>,
  },
  tags: {
    all: () => AxiosPromise<TagsRaw>,
  },
  vacancies: {
    all: () => AxiosPromise<VacanciesRaw>,
    myVacancies: () => AxiosPromise<VacanciesRaw>,
    withStages: (stage: string) => AxiosPromise<VacanciesRaw>,
    changeState: (id: number, state: string) => AxiosPromise<VacancyRaw>,
    findBy: (params: { vacancy: * }) => AxiosPromise<VacancyRaw>,
    update: (id: number, params: { vacancy: * }) => AxiosPromise<VacancyRaw>,
    findById: (id: number) => AxiosPromise<VacancyRaw>,
  },
  vacancy_stage_group: () => AxiosPromise<MilestonesGroupRaw>,
  template_stages: () => AxiosPromise<MilestoneTemplateRaw>,
  resume_sources: () => AxiosPromise<Array<ResumeSourceRaw>>,
  languages: () => AxiosPromise<Array<LanguageRaw>>,
  languagesLevel: () => AxiosPromise<Array<LanguageLevelRaw>>,
  educationLevel: () => AxiosPromise<Array<EducationLevelRaw>>,
  departments: {
    tree: () => AxiosPromise<DepartmentsRaw>,
  },
  services: {
    all: () => AxiosPromise<ServicesRaw>,
    find: () => AxiosPromise<ServiceRaw>,
  },
  bids: {
    all: () => AxiosPromise<BidsRaw>,
    find: () => AxiosPromise<BidRaw>,
  },
}
