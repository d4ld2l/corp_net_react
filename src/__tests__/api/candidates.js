import api from '../../api'

describe('API Candidates', () => {
  it('should fetch all candidates', async () => {
    const response = await api.candidates.all()
    expect(response.data).toMatchSnapshot()
  })

  it('should find candidate by id', async () => {
    const response = await api.candidates.findById(1)
    expect(response.data).toMatchSnapshot()
  })
})
