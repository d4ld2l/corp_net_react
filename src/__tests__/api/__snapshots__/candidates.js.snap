// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API Candidates should fetch all candidates 1`] = `
Array [
  Object {
    "birthdate": null,
    "candidate_vacancies": Array [
      Object {
        "candidate_id": 1,
        "created_at": "2017-08-11T18:21:18.446Z",
        "current_vacancy_stage_id": 16,
        "id": 1,
        "updated_at": "2017-08-11T18:21:18.446Z",
        "vacancy_id": 43,
      },
      Object {
        "candidate_id": 1,
        "created_at": "2017-08-14T14:13:47.194Z",
        "current_vacancy_stage_id": 56,
        "id": 5,
        "updated_at": "2017-08-17T11:24:27.863Z",
        "vacancy_id": 48,
      },
    ],
    "created_at": "2017-08-10T08:45:27.030Z",
    "first_name": "Откорней",
    "id": 1,
    "last_name": "Докончиков",
    "middle_name": "Муфлонович",
    "updated_at": "2017-08-10T08:45:27.030Z",
    "vacancy_id": null,
  },
  Object {
    "birthdate": "1989-02-28",
    "candidate_vacancies": Array [],
    "created_at": "2017-08-10T09:38:52.457Z",
    "first_name": null,
    "id": 2,
    "last_name": null,
    "middle_name": null,
    "updated_at": "2017-08-10T09:38:52.457Z",
    "vacancy_id": null,
  },
  Object {
    "birthdate": "1994-11-17",
    "candidate_vacancies": Array [
      Object {
        "candidate_id": 3,
        "created_at": "2017-08-11T19:04:17.983Z",
        "current_vacancy_stage_id": 16,
        "id": 3,
        "updated_at": "2017-08-11T19:04:17.983Z",
        "vacancy_id": 43,
      },
      Object {
        "candidate_id": 3,
        "created_at": "2017-08-14T14:14:30.730Z",
        "current_vacancy_stage_id": 63,
        "id": 6,
        "updated_at": "2017-08-17T10:11:30.822Z",
        "vacancy_id": 50,
      },
    ],
    "created_at": "2017-08-10T09:54:52.718Z",
    "first_name": "Виктория",
    "id": 3,
    "last_name": "Сабурова",
    "middle_name": "Игоревна",
    "updated_at": "2017-08-10T09:54:52.718Z",
    "vacancy_id": null,
  },
  Object {
    "birthdate": null,
    "candidate_vacancies": Array [
      Object {
        "candidate_id": 4,
        "created_at": "2017-08-15T05:55:47.507Z",
        "current_vacancy_stage_id": 77,
        "id": 9,
        "updated_at": "2017-08-15T05:55:47.507Z",
        "vacancy_id": 52,
      },
    ],
    "created_at": "2017-08-10T11:07:48.986Z",
    "first_name": "Виктория",
    "id": 4,
    "last_name": "Сабурова",
    "middle_name": "Игоревна",
    "updated_at": "2017-08-10T11:07:48.986Z",
    "vacancy_id": null,
  },
  Object {
    "birthdate": "2017-08-02",
    "candidate_vacancies": Array [
      Object {
        "candidate_id": 5,
        "created_at": "2017-08-14T18:55:33.443Z",
        "current_vacancy_stage_id": 77,
        "id": 8,
        "updated_at": "2017-08-14T18:55:33.443Z",
        "vacancy_id": 52,
      },
    ],
    "created_at": "2017-08-10T14:01:10.560Z",
    "first_name": "Откорней",
    "id": 5,
    "last_name": "Докончиков",
    "middle_name": "Натальевич",
    "resume": Object {
      "additional_contacts": Array [],
      "birthdate": null,
      "candidate_id": 5,
      "city": null,
      "comment": null,
      "created_at": "2017-10-16T08:51:55.170Z",
      "desired_position": null,
      "documents": Array [],
      "education_level_id": null,
      "email": null,
      "employment_type": null,
      "experience": null,
      "first_name": null,
      "have_children": null,
      "id": 22,
      "language_skills": Array [],
      "last_name": null,
      "martial_condition": null,
      "middle_name": null,
      "phone": null,
      "photo": Object {
        "url": null,
      },
      "position": null,
      "preferred_contact_type": null,
      "professional_specialization_id": null,
      "resume_recommendations": Array [],
      "resume_work_experiences": Array [],
      "salary_level": null,
      "sex": null,
      "skills": Array [
        Object {
          "id": 66,
          "name": "бегать",
          "taggings_count": 1,
        },
        Object {
          "id": 67,
          "name": "прыгать",
          "taggings_count": 1,
        },
      ],
      "skills_description": null,
      "skype": null,
      "updated_at": "2017-10-16T08:51:55.170Z",
      "user_id": null,
      "vacancy_id": null,
      "working_schedule": null,
    },
    "updated_at": "2017-08-10T14:01:10.560Z",
    "vacancy_id": null,
  },
  Object {
    "birthdate": "2017-08-05",
    "candidate_vacancies": Array [
      Object {
        "candidate_id": 6,
        "created_at": "2017-08-17T09:21:09.923Z",
        "current_vacancy_stage_id": 147,
        "id": 14,
        "updated_at": "2017-08-17T09:21:09.923Z",
        "vacancy_id": 65,
      },
      Object {
        "candidate_id": 6,
        "created_at": "2017-08-15T09:52:52.991Z",
        "current_vacancy_stage_id": 53,
        "id": 10,
        "updated_at": "2017-09-26T12:19:50.970Z",
        "vacancy_id": 48,
      },
    ],
    "created_at": "2017-08-10T15:54:52.200Z",
    "first_name": "Рулон",
    "id": 6,
    "last_name": "Обоев",
    "middle_name": "Папуасович",
    "resume": Object {
      "additional_contacts": Array [],
      "birthdate": null,
      "candidate_id": 6,
      "city": "moscow",
      "comment": null,
      "created_at": "2017-08-10T15:54:52.203Z",
      "desired_position": null,
      "documents": Array [],
      "education_level_id": null,
      "email": "mumu@milk.com",
      "employment_type": null,
      "experience": null,
      "first_name": null,
      "have_children": null,
      "id": 5,
      "language_skills": Array [],
      "last_name": null,
      "martial_condition": null,
      "middle_name": null,
      "phone": "+7 (123) 456 78 90",
      "photo": Object {
        "url": "/uploads/resume/photo/5/photo.jpeg",
      },
      "position": "Инструктор по",
      "preferred_contact_type": "email",
      "professional_specialization_id": null,
      "resume_recommendations": Array [],
      "resume_work_experiences": Array [],
      "salary_level": null,
      "sex": null,
      "skills": Array [],
      "skills_description": null,
      "skype": "qwere",
      "updated_at": "2017-08-10T15:54:52.203Z",
      "user_id": null,
      "vacancy_id": null,
      "working_schedule": null,
    },
    "updated_at": "2017-08-10T15:54:52.200Z",
    "vacancy_id": null,
  },
  Object {
    "birthdate": "1989-02-28",
    "candidate_vacancies": Array [
      Object {
        "candidate_id": 7,
        "created_at": "2017-08-11T18:38:11.258Z",
        "current_vacancy_stage_id": 16,
        "id": 2,
        "updated_at": "2017-08-11T18:38:11.258Z",
        "vacancy_id": 43,
      },
      Object {
        "candidate_id": 7,
        "created_at": "2017-08-14T10:18:04.591Z",
        "current_vacancy_stage_id": 70,
        "id": 4,
        "updated_at": "2017-08-17T10:13:29.140Z",
        "vacancy_id": 50,
      },
      Object {
        "candidate_id": 7,
        "created_at": "2017-08-15T10:26:22.954Z",
        "current_vacancy_stage_id": 143,
        "id": 11,
        "updated_at": "2017-08-17T11:15:47.523Z",
        "vacancy_id": 64,
      },
      Object {
        "candidate_id": 7,
        "created_at": "2017-08-14T18:54:58.781Z",
        "current_vacancy_stage_id": 52,
        "id": 7,
        "updated_at": "2017-08-21T13:51:09.404Z",
        "vacancy_id": 48,
      },
      Object {
        "candidate_id": 7,
        "created_at": "2017-10-12T12:47:01.495Z",
        "current_vacancy_stage_id": 220,
        "id": 24,
        "updated_at": "2017-10-12T12:47:01.495Z",
        "vacancy_id": 90,
      },
    ],
    "created_at": "2017-08-11T07:44:15.489Z",
    "first_name": "Дмитрий",
    "id": 7,
    "last_name": "Мишуков",
    "middle_name": "Михайлович",
    "resume": Object {
      "additional_contacts": Array [],
      "birthdate": null,
      "candidate_id": 7,
      "city": "moscow",
      "comment": null,
      "created_at": "2017-08-11T07:44:15.491Z",
      "desired_position": null,
      "documents": Array [],
      "education_level_id": null,
      "email": null,
      "employment_type": null,
      "experience": null,
      "first_name": null,
      "have_children": null,
      "id": 6,
      "language_skills": Array [],
      "last_name": null,
      "martial_condition": null,
      "middle_name": null,
      "phone": "+7(965) 277-91-37",
      "photo": Object {
        "url": "/uploads/resume/photo/6/photo.jpeg",
      },
      "position": "Java разработчик",
      "preferred_contact_type": "phone",
      "professional_specialization_id": null,
      "resume_recommendations": Array [],
      "resume_work_experiences": Array [],
      "salary_level": null,
      "sex": null,
      "skills": Array [],
      "skills_description": null,
      "skype": null,
      "updated_at": "2017-08-11T07:44:15.491Z",
      "user_id": null,
      "vacancy_id": null,
      "working_schedule": null,
    },
    "updated_at": "2017-08-11T07:44:15.489Z",
    "vacancy_id": null,
  },
  Object {
    "birthdate": "2017-08-04",
    "candidate_vacancies": Array [],
    "created_at": "2017-08-14T18:54:23.532Z",
    "first_name": "йц",
    "id": 8,
    "last_name": "Докончиков",
    "middle_name": "hgfd",
    "resume": Object {
      "additional_contacts": Array [],
      "birthdate": null,
      "candidate_id": 8,
      "city": "st-peterburg",
      "comment": null,
      "created_at": "2017-08-14T18:54:23.534Z",
      "desired_position": null,
      "documents": Array [],
      "education_level_id": null,
      "email": "root@e.co",
      "employment_type": null,
      "experience": null,
      "first_name": null,
      "have_children": null,
      "id": 7,
      "language_skills": Array [],
      "last_name": null,
      "martial_condition": null,
      "middle_name": null,
      "phone": "+7 (123) 456 78 90",
      "photo": Object {
        "url": "/uploads/resume/photo/7/photo.jpeg",
      },
      "position": "hgfd",
      "preferred_contact_type": "phone",
      "professional_specialization_id": null,
      "resume_recommendations": Array [],
      "resume_work_experiences": Array [],
      "salary_level": null,
      "sex": null,
      "skills": Array [],
      "skills_description": null,
      "skype": "gf",
      "updated_at": "2017-08-14T18:54:23.534Z",
      "user_id": null,
      "vacancy_id": null,
      "working_schedule": null,
    },
    "updated_at": "2017-08-14T18:54:23.532Z",
    "vacancy_id": null,
  },
  Object {
    "birthdate": "2017-08-25",
    "candidate_vacancies": Array [],
    "created_at": "2017-08-15T18:20:45.347Z",
    "first_name": "Откорней",
    "id": 9,
    "last_name": "Докончиков",
    "middle_name": null,
    "resume": Object {
      "additional_contacts": Array [],
      "birthdate": null,
      "candidate_id": 9,
      "city": "Москва",
      "comment": null,
      "created_at": "2017-08-15T18:20:45.349Z",
      "desired_position": null,
      "documents": Array [],
      "education_level_id": null,
      "email": "rrr@ww.ru",
      "employment_type": null,
      "experience": null,
      "first_name": null,
      "have_children": null,
      "id": 8,
      "language_skills": Array [],
      "last_name": null,
      "martial_condition": null,
      "middle_name": null,
      "phone": "+7 (123) 456 78 90",
      "photo": Object {
        "url": null,
      },
      "position": "Dancer",
      "preferred_contact_type": "phone",
      "professional_specialization_id": null,
      "resume_recommendations": Array [],
      "resume_work_experiences": Array [
        Object {
          "company_name": null,
          "created_at": "2017-08-15T18:20:45.351Z",
          "end_date": null,
          "experience_description": null,
          "id": 2,
          "position": "Dancer",
          "region": null,
          "resume_id": 8,
          "start_date": null,
          "updated_at": "2017-08-15T18:20:45.351Z",
          "website": null,
        },
      ],
      "salary_level": null,
      "sex": null,
      "skills": Array [],
      "skills_description": null,
      "skype": null,
      "updated_at": "2017-08-15T18:20:45.349Z",
      "user_id": null,
      "vacancy_id": null,
      "working_schedule": null,
    },
    "updated_at": "2017-08-15T18:20:45.347Z",
    "vacancy_id": null,
  },
  Object {
    "birthdate": "1989-02-23",
    "candidate_vacancies": Array [
      Object {
        "candidate_id": 10,
        "created_at": "2017-08-16T05:38:40.837Z",
        "current_vacancy_stage_id": 50,
        "id": 12,
        "updated_at": "2017-08-17T11:14:49.716Z",
        "vacancy_id": 48,
      },
    ],
    "created_at": "2017-08-15T18:26:58.140Z",
    "first_name": "Лучано",
    "id": 10,
    "last_name": "Паперони",
    "middle_name": null,
    "resume": Object {
      "additional_contacts": Array [],
      "birthdate": null,
      "candidate_id": 10,
      "city": "Санкт-Петербург",
      "comment": null,
      "created_at": "2017-08-15T18:26:58.142Z",
      "desired_position": null,
      "documents": Array [],
      "education_level_id": null,
      "email": "pizza@fake.com",
      "employment_type": null,
      "experience": null,
      "first_name": null,
      "have_children": null,
      "id": 9,
      "language_skills": Array [],
      "last_name": null,
      "martial_condition": null,
      "middle_name": null,
      "phone": "+7 (123) 456 78 90",
      "photo": Object {
        "url": "/uploads/resume/photo/9/photo.jpeg",
      },
      "position": "Мастер пиццы",
      "preferred_contact_type": "email",
      "professional_specialization_id": null,
      "resume_recommendations": Array [],
      "resume_work_experiences": Array [
        Object {
          "company_name": null,
          "created_at": "2017-08-15T18:26:58.144Z",
          "end_date": null,
          "experience_description": null,
          "id": 3,
          "position": "Мастер пиццы",
          "region": null,
          "resume_id": 9,
          "start_date": null,
          "updated_at": "2017-08-15T18:26:58.144Z",
          "website": null,
        },
      ],
      "salary_level": null,
      "sex": null,
      "skills": Array [],
      "skills_description": null,
      "skype": null,
      "updated_at": "2017-08-15T18:26:58.142Z",
      "user_id": null,
      "vacancy_id": null,
      "working_schedule": null,
    },
    "updated_at": "2017-08-15T18:26:58.140Z",
    "vacancy_id": null,
  },
  Object {
    "birthdate": "1988-08-11",
    "candidate_vacancies": Array [
      Object {
        "candidate_id": 11,
        "created_at": "2017-08-16T15:24:06.103Z",
        "current_vacancy_stage_id": 54,
        "id": 13,
        "updated_at": "2017-08-17T12:04:03.591Z",
        "vacancy_id": 48,
      },
      Object {
        "candidate_id": 11,
        "created_at": "2017-08-22T07:38:23.295Z",
        "current_vacancy_stage_id": 167,
        "id": 17,
        "updated_at": "2017-08-22T07:38:23.295Z",
        "vacancy_id": 71,
      },
    ],
    "created_at": "2017-08-16T15:23:20.183Z",
    "first_name": "Степан",
    "id": 11,
    "last_name": "Гирин",
    "middle_name": "Владимирович",
    "resume": Object {
      "additional_contacts": Array [],
      "birthdate": null,
      "candidate_id": 11,
      "city": "Москва",
      "comment": null,
      "created_at": "2017-08-16T15:23:20.186Z",
      "desired_position": null,
      "documents": Array [],
      "education_level_id": null,
      "email": "mishukov89@gmail.com",
      "employment_type": null,
      "experience": null,
      "first_name": null,
      "have_children": null,
      "id": 10,
      "language_skills": Array [],
      "last_name": null,
      "martial_condition": null,
      "middle_name": null,
      "phone": "+79161382283",
      "photo": Object {
        "url": null,
      },
      "position": "Менеджер",
      "preferred_contact_type": "phone",
      "professional_specialization_id": null,
      "resume_recommendations": Array [],
      "resume_work_experiences": Array [
        Object {
          "company_name": null,
          "created_at": "2017-08-16T15:23:20.189Z",
          "end_date": null,
          "experience_description": null,
          "id": 4,
          "position": "Менеджер",
          "region": null,
          "resume_id": 10,
          "start_date": null,
          "updated_at": "2017-08-16T15:23:20.189Z",
          "website": null,
        },
      ],
      "salary_level": null,
      "sex": null,
      "skills": Array [],
      "skills_description": null,
      "skype": null,
      "updated_at": "2017-08-16T15:23:20.186Z",
      "user_id": null,
      "vacancy_id": null,
      "working_schedule": null,
    },
    "updated_at": "2017-08-16T15:23:20.183Z",
    "vacancy_id": null,
  },
  Object {
    "birthdate": "1981-08-08",
    "candidate_vacancies": Array [],
    "created_at": "2017-08-17T09:30:52.329Z",
    "first_name": "Первый",
    "id": 12,
    "last_name": "Иванов",
    "middle_name": "Иванович",
    "resume": Object {
      "additional_contacts": Array [],
      "birthdate": null,
      "candidate_id": 12,
      "city": "Москва",
      "comment": null,
      "created_at": "2017-08-17T09:30:52.331Z",
      "desired_position": null,
      "documents": Array [],
      "education_level_id": null,
      "email": "ii@ya.ru",
      "employment_type": null,
      "experience": null,
      "first_name": null,
      "have_children": null,
      "id": 11,
      "language_skills": Array [],
      "last_name": null,
      "martial_condition": null,
      "middle_name": null,
      "phone": "+79051111111",
      "photo": Object {
        "url": null,
      },
      "position": "Разработчик",
      "preferred_contact_type": "phone",
      "professional_specialization_id": null,
      "resume_recommendations": Array [],
      "resume_work_experiences": Array [
        Object {
          "company_name": null,
          "created_at": "2017-08-17T09:30:52.333Z",
          "end_date": null,
          "experience_description": null,
          "id": 5,
          "position": "Разработчик",
          "region": null,
          "resume_id": 11,
          "start_date": null,
          "updated_at": "2017-08-17T09:30:52.333Z",
          "website": null,
        },
      ],
      "salary_level": null,
      "sex": null,
      "skills": Array [],
      "skills_description": null,
      "skype": null,
      "updated_at": "2017-08-17T09:30:52.331Z",
      "user_id": null,
      "vacancy_id": null,
      "working_schedule": null,
    },
    "updated_at": "2017-08-17T09:30:52.329Z",
    "vacancy_id": null,
  },
  Object {
    "birthdate": "1982-08-08",
    "candidate_vacancies": Array [
      Object {
        "candidate_id": 13,
        "created_at": "2017-08-22T07:38:49.924Z",
        "current_vacancy_stage_id": 167,
        "id": 18,
        "updated_at": "2017-08-22T07:38:49.924Z",
        "vacancy_id": 71,
      },
      Object {
        "candidate_id": 13,
        "created_at": "2017-10-12T07:37:20.340Z",
        "current_vacancy_stage_id": 157,
        "id": 23,
        "updated_at": "2017-10-12T07:37:20.340Z",
        "vacancy_id": 69,
      },
    ],
    "created_at": "2017-08-17T11:21:41.729Z",
    "first_name": "Второй",
    "id": 13,
    "last_name": "Иванов",
    "middle_name": "Иванович",
    "resume": Object {
      "additional_contacts": Array [],
      "birthdate": null,
      "candidate_id": 13,
      "city": "Москва",
      "comment": null,
      "created_at": "2017-08-17T11:21:41.731Z",
      "desired_position": null,
      "documents": Array [],
      "education_level_id": null,
      "email": "ii2@ya.ru",
      "employment_type": null,
      "experience": null,
      "first_name": null,
      "have_children": null,
      "id": 12,
      "language_skills": Array [],
      "last_name": null,
      "martial_condition": null,
      "middle_name": null,
      "phone": "+79032222222",
      "photo": Object {
        "url": "/uploads/resume/photo/12/photo.png",
      },
      "position": "Аналитик",
      "preferred_contact_type": "phone",
      "professional_specialization_id": null,
      "resume_recommendations": Array [],
      "resume_work_experiences": Array [
        Object {
          "company_name": null,
          "created_at": "2017-08-17T11:21:41.733Z",
          "end_date": null,
          "experience_description": null,
          "id": 6,
          "position": "Аналитик",
          "region": null,
          "resume_id": 12,
          "start_date": null,
          "updated_at": "2017-08-17T11:21:41.733Z",
          "website": null,
        },
      ],
      "salary_level": null,
      "sex": "male",
      "skills": Array [],
      "skills_description": null,
      "skype": "ii2",
      "updated_at": "2017-08-17T11:21:41.731Z",
      "user_id": null,
      "vacancy_id": null,
      "working_schedule": null,
    },
    "updated_at": "2017-08-17T11:21:41.729Z",
    "vacancy_id": null,
  },
  Object {
    "birthdate": "2017-08-03",
    "candidate_vacancies": Array [
      Object {
        "candidate_id": 14,
        "created_at": "2017-08-17T11:32:54.218Z",
        "current_vacancy_stage_id": 62,
        "id": 15,
        "updated_at": "2017-08-17T11:32:54.218Z",
        "vacancy_id": 50,
      },
    ],
    "created_at": "2017-08-17T11:32:54.206Z",
    "first_name": "Откорней",
    "id": 14,
    "last_name": "Докончиков",
    "middle_name": null,
    "resume": Object {
      "additional_contacts": Array [],
      "birthdate": null,
      "candidate_id": 14,
      "city": "Санкт-Петербург",
      "comment": null,
      "created_at": "2017-08-17T11:32:54.208Z",
      "desired_position": null,
      "documents": Array [],
      "education_level_id": null,
      "email": "12@ss.ru",
      "employment_type": null,
      "experience": null,
      "first_name": null,
      "have_children": null,
      "id": 13,
      "language_skills": Array [],
      "last_name": null,
      "martial_condition": null,
      "middle_name": null,
      "phone": "+7 (123) 456 78 90",
      "photo": Object {
        "url": null,
      },
      "position": "sdfdsfds",
      "preferred_contact_type": "email",
      "professional_specialization_id": null,
      "resume_recommendations": Array [],
      "resume_work_experiences": Array [
        Object {
          "company_name": null,
          "created_at": "2017-08-17T11:32:54.210Z",
          "end_date": null,
          "experience_description": null,
          "id": 7,
          "position": "sdfdsfds",
          "region": null,
          "resume_id": 13,
          "start_date": null,
          "updated_at": "2017-08-17T11:32:54.210Z",
          "website": null,
        },
      ],
      "salary_level": null,
      "sex": null,
      "skills": Array [],
      "skills_description": null,
      "skype": null,
      "updated_at": "2017-08-17T11:32:54.208Z",
      "user_id": null,
      "vacancy_id": null,
      "working_schedule": null,
    },
    "updated_at": "2017-08-17T11:32:54.206Z",
    "vacancy_id": null,
  },
  Object {
    "birthdate": "1983-08-08",
    "candidate_vacancies": Array [
      Object {
        "candidate_id": 15,
        "created_at": "2017-08-22T07:39:02.311Z",
        "current_vacancy_stage_id": 169,
        "id": 19,
        "updated_at": "2017-08-22T07:40:26.669Z",
        "vacancy_id": 71,
      },
      Object {
        "candidate_id": 15,
        "created_at": "2017-08-17T11:54:09.673Z",
        "current_vacancy_stage_id": 159,
        "id": 16,
        "updated_at": "2017-08-30T06:31:43.673Z",
        "vacancy_id": 69,
      },
    ],
    "created_at": "2017-08-17T11:54:09.658Z",
    "first_name": "Третий",
    "id": 15,
    "last_name": "Иванов",
    "middle_name": "Иванович",
    "resume": Object {
      "additional_contacts": Array [],
      "birthdate": null,
      "candidate_id": 15,
      "city": "Москва",
      "comment": null,
      "created_at": "2017-08-17T11:54:09.660Z",
      "desired_position": null,
      "documents": Array [],
      "education_level_id": null,
      "email": "ii3@ya.ru",
      "employment_type": null,
      "experience": null,
      "first_name": null,
      "have_children": null,
      "id": 14,
      "language_skills": Array [],
      "last_name": null,
      "martial_condition": null,
      "middle_name": null,
      "phone": "+79053333333",
      "photo": Object {
        "url": "/uploads/resume/photo/14/photo.png",
      },
      "position": "Аналитик",
      "preferred_contact_type": "phone",
      "professional_specialization_id": null,
      "resume_recommendations": Array [],
      "resume_work_experiences": Array [
        Object {
          "company_name": null,
          "created_at": "2017-08-17T11:54:09.662Z",
          "end_date": null,
          "experience_description": null,
          "id": 8,
          "position": "Аналитик",
          "region": null,
          "resume_id": 14,
          "start_date": null,
          "updated_at": "2017-08-17T11:54:09.662Z",
          "website": null,
        },
      ],
      "salary_level": null,
      "sex": "male",
      "skills": Array [],
      "skills_description": null,
      "skype": "ii3",
      "updated_at": "2017-08-17T11:54:09.660Z",
      "user_id": null,
      "vacancy_id": null,
      "working_schedule": null,
    },
    "updated_at": "2017-08-17T11:54:09.658Z",
    "vacancy_id": null,
  },
  Object {
    "birthdate": "2017-09-26",
    "candidate_vacancies": Array [],
    "created_at": "2017-09-26T13:19:12.565Z",
    "first_name": null,
    "id": 16,
    "last_name": null,
    "middle_name": null,
    "resume": Object {
      "additional_contacts": Array [],
      "birthdate": null,
      "candidate_id": 16,
      "city": "Москва",
      "comment": null,
      "created_at": "2017-09-26T13:19:12.569Z",
      "desired_position": null,
      "documents": Array [],
      "education_level_id": null,
      "email": null,
      "employment_type": null,
      "experience": null,
      "first_name": null,
      "have_children": null,
      "id": 19,
      "language_skills": Array [],
      "last_name": null,
      "martial_condition": null,
      "middle_name": null,
      "phone": null,
      "photo": Object {
        "url": null,
      },
      "position": "ппп",
      "preferred_contact_type": "phone",
      "professional_specialization_id": null,
      "resume_recommendations": Array [],
      "resume_work_experiences": Array [],
      "salary_level": null,
      "sex": null,
      "skills": Array [],
      "skills_description": null,
      "skype": null,
      "updated_at": "2017-09-26T13:19:12.569Z",
      "user_id": null,
      "vacancy_id": null,
      "working_schedule": null,
    },
    "updated_at": "2017-09-26T13:19:12.565Z",
    "vacancy_id": null,
  },
]
`;

exports[`API Candidates should find candidate by id 1`] = `
Object {
  "birthdate": null,
  "candidate_vacancies": Array [
    Object {
      "candidate_id": 1,
      "created_at": "2017-08-11T18:21:18.446Z",
      "current_vacancy_stage_id": 16,
      "id": 1,
      "updated_at": "2017-08-11T18:21:18.446Z",
      "vacancy_id": 43,
    },
    Object {
      "candidate_id": 1,
      "created_at": "2017-08-14T14:13:47.194Z",
      "current_vacancy_stage_id": 56,
      "id": 5,
      "updated_at": "2017-08-17T11:24:27.863Z",
      "vacancy_id": 48,
    },
  ],
  "created_at": "2017-08-10T08:45:27.030Z",
  "first_name": "Откорней",
  "id": 1,
  "last_name": "Докончиков",
  "middle_name": "Муфлонович",
  "updated_at": "2017-08-10T08:45:27.030Z",
  "vacancy_id": null,
}
`;
